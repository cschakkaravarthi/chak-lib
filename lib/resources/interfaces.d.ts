/// <reference types="react" />
export interface MinimalCardProps {
    MainLink?: React.ComponentType;
    mainLinkText: string;
    extension?: string;
    external?: boolean;
}
//# sourceMappingURL=interfaces.d.ts.map