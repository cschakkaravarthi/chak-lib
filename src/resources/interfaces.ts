export interface MinimalCardProps {
  MainLink?: React.ComponentType;
  mainLinkText: string;
  extension?: string;
  external?: boolean;
}
