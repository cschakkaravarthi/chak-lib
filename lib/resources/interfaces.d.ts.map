{"version":3,"file":"interfaces.d.ts","sourceRoot":"","sources":["../../src/resources/interfaces.ts"],"names":[],"mappings":";AAAA,MAAM,WAAW,gBAAgB;IAC/B,QAAQ,CAAC,EAAE,KAAK,CAAC,aAAa,CAAC;IAC/B,YAAY,EAAE,MAAM,CAAC;IACrB,SAAS,CAAC,EAAE,MAAM,CAAC;IACnB,QAAQ,CAAC,EAAE,OAAO,CAAC;CACpB"}