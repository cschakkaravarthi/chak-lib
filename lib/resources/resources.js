"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
exports.__esModule = true;
var react_1 = __importDefault(require("react"));
var Card_1 = __importDefault(require("react-bootstrap/Card"));
var react_bootstrap_1 = require("react-bootstrap");
var images_1 = require("../assets/images");
exports.Resources = function (props) {
    var _a = props.MainLink, MainLink = _a === void 0 ? null : _a, mainLinkText = props.mainLinkText, extension = props.extension, _b = props.external, external = _b === void 0 ? !extension : _b;
    return (react_1["default"].createElement(Card_1["default"], { className: 'rounded-0 p-2' }, MainLink && (react_1["default"].createElement(MainLink, null,
        react_1["default"].createElement(react_bootstrap_1.Row, { className: 'w-100 d-flex justify-content-between m-0 p-0 align-items-center' },
            react_1["default"].createElement(react_bootstrap_1.Col, { sm: 9, md: 9, lg: 9, xl: 9 },
                react_1["default"].createElement("strong", { className: 'font-size-sm' }, mainLinkText)),
            react_1["default"].createElement(react_bootstrap_1.Col, { sm: 3, md: 3, lg: 3, xl: 3, className: 'text-right' },
                extension && (react_1["default"].createElement("div", { className: 'pr-1 gray-600 d-inline text-uppercase text-right text-nowrap font-size-sm' },
                    react_1["default"].createElement("strong", null, extension))),
                external && (react_1["default"].createElement("img", { src: images_1.images.external, width: 20, height: 20 }))))))));
};
//# sourceMappingURL=resources.js.map