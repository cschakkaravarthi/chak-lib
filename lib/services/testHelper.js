"use strict";
/* eslint @typescript-eslint/no-explicit-any: "off" */
// In this one file we do actually want to use the "any" type. This is a very generic function.
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
exports.__esModule = true;
var shallow_1 = __importDefault(require("react-test-renderer/shallow"));
exports.shallowRender = function (toRender) {
    var renderer = shallow_1["default"].createRenderer();
    renderer.render(toRender);
    return renderer.getRenderOutput();
};
//# sourceMappingURL=testHelper.js.map