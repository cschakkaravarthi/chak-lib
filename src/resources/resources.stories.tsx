import React from 'react';
import StoriesContainer from '../StoriesContainer';
import { Resources } from './resources';

export default {
  component: Resources,
  title: 'Resources'
};

export const resources = (): JSX.Element => (
  <StoriesContainer>
    <Resources
      mainLinkText="ReallySuperExtraLongNameForTheResource with spaces even bigger"
      MainLink={p => <a href="#">{p.children}</a>}
      extension="PPT"
      external={true}
    />
  </StoriesContainer>
);
