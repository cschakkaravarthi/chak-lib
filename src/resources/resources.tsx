import React, { FC } from 'react';
import SourceCard from 'react-bootstrap/Card';
import { Row, Col } from 'react-bootstrap';
import { MinimalCardProps } from './interfaces';
import { images } from '../assets/images';

export const Resources: FC<MinimalCardProps> = props => {
  const { MainLink = null, mainLinkText, extension, external = !extension } = props;

  return (
    <SourceCard className="rounded-0 p-2">
      {MainLink && (
        <MainLink>
          <Row className="w-100 d-flex justify-content-between m-0 p-0 align-items-center">
            <Col sm={9} md={9} lg={9} xl={9}>
              <strong className="font-size-sm">{mainLinkText}</strong>
            </Col>
            <Col sm={3} md={3} lg={3} xl={3} className="text-right">
              {extension && (
                <div className="pr-1 gray-600 d-inline text-uppercase text-right text-nowrap font-size-sm">
                  <strong>{extension}</strong>
                </div>
              )}
              {external && <img src={images.external} width={20} height={20} />}
            </Col>
          </Row>
        </MainLink>
      )}
    </SourceCard>
  );
};
