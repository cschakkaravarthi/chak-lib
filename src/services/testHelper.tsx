/* eslint @typescript-eslint/no-explicit-any: "off" */
// In this one file we do actually want to use the "any" type. This is a very generic function.

import ShallowRenderer from 'react-test-renderer/shallow';

export const shallowRender = (toRender: any): Record<string, any> => {
  const renderer = ShallowRenderer.createRenderer();
  renderer.render(toRender);
  return renderer.getRenderOutput();
};
