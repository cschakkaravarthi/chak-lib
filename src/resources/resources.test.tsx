import React from 'react';

import { shallowRender } from '../services/testHelper';
import { Resources } from './resources';

describe('Resources', () => {
  it('should render correctly', () => {
    const tree = shallowRender(
      <Resources
        mainLinkText="Link Text"
        MainLink={p => <a href="#">{p.children}</a>}
      />
    );
    expect(tree).toMatchSnapshot();
  });
});
