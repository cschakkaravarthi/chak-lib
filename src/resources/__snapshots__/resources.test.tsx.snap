// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Resources should render correctly 1`] = `
<ForwardRef
  body={false}
  className="rounded-0 p-2"
>
  <MainLink>
    <ForwardRef
      className="w-100 d-flex justify-content-between m-0 p-0 align-items-center"
      noGutters={false}
    >
      <ForwardRef
        lg={9}
        md={9}
        sm={9}
        xl={9}
      >
        <strong
          className="font-size-sm"
        >
          Link Text
        </strong>
      </ForwardRef>
      <ForwardRef
        className="text-right"
        lg={3}
        md={3}
        sm={3}
        xl={3}
      >
        <img
          height={20}
          src="test-file-stub"
          width={20}
        />
      </ForwardRef>
    </ForwardRef>
  </MainLink>
</ForwardRef>
`;
